pocs/camera/zwo.py

```diff
@@ -0,0 +1,446 @@
+                    logger.warning("Found ZWO ASI camera with no ID set")
+                    break
+                assert id.endswith(')'), self.logger.error("Expected ID enclosed in ()")
+                id = id[:-1]
+                Camera._ids.append(id)
+
+            logger.debug('Connected ASI ZWO cameras: {}'.format(Camera._ids))
+
+        try:
+            self._camera_index = Camera._ids.index(serial_number)
+        except ValueError:
+            msg = "Could not find ZWO ASI camera with ID '{}'".format(serial_number)
+            logger.error(msg)
+            warn(msg)
+            return
+        else:
+            logger.debug("Found ZWO ASI Camera with ID '{}' at index {}".format(
+                serial_number, self._camera_index))
+
+        if serial_number in Camera._assigned_ids:
+            msg = "ZWO ASI Camera with ID '{}' already in use".format(serial_number)
+            logger.error(msg)
+            warn(msg)
+            return
+
+        super().__init__(name, *args, **kwargs)
+        self.connect()
+        assert self.is_connected, error.PanError("Could not connect to {}".format(self))
+
+        Camera._assigned_ids.append(self.uid)
+
+        self._setter_try('ccd_set_point', 'set_point', set_point)
+
+        if filter_type:
+            # connect() will have set this based on camera info, but that doesn't know about filters
+            # upstream of the CCD. Can be set manually here, or handled by a filterwheel attribute.
+            self._filter_type = filter_type
+
+        self._setter_try('gain', 'gain', gain)
+
+        if image_type:
+            self._setter_try('image_type', 'image_type', image_type)
+        else:
+            # Take monochrome 12 bit raw images by default, if we can
+            if 'RAW16' in self._info['supported_video_format']:
+                self.image_type = 'RAW16'
+
+        self.logger.info('{} initialised'.format(self))
+
+    def __del__(self):
+        """ Attempt some clean up """
+        with suppress(AttributeError):
+            id = self.uid
+            Camera._assigned_ids.remove(id)
+            self.logger.debug('Removed {} from assigned IDs list'.format(id))
+        with suppress(AttributeError):
+            camera_ID = self._camera_ID
+            Camera._ASIDriver.close_camera(camera_ID)
+            self.logger.debug("Closed ZWO camera {}".format(camera_ID))
+
+    # Properties
+
+    @AbstractCamera.uid.getter
+    def uid(self):
+        """Return unique identifier for camera.
+
+        Need to overide this because the base class only returns the 1st
+        6 characters of the serial number, which is not a unique identifier
+        for most of the camera types
+        """
+        return self._serial_number
+
+    @property
+    def image_type(self):
+        """ Current camera image type, one of 'RAW8', 'RAW16', 'Y8', 'RGB24' """
+        roi_format = Camera._ASIDriver.get_roi_format(self._camera_ID)
+        return roi_format['image_type']
```

Comment:
Is there a default `image_type`? Regardless, the setter can fail below if someone tries to set an unsupported type, which would then cause this property to throw a `KeyError` if accessed. Not sure if that is the intent.

Reply:
Yes, `image_type` is an internal parameter of the camera. It has a default value and it will always have a value (internally it's just an integer at a certain memory location).  A failed attempt to set it to an unsupported value will just leave it set to whatever the current value is.